spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/types/ColumnOrValue.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.errorhandling.types
+
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{array, col, lit, map_from_arrays, typedLit}
+import org.apache.spark.sql.types.{DataType, StringType}
+import za.co.absa.spark.commons.sql.functions.null_col
+
+import scala.language.higherKinds
+
+trait ColumnOrValue[T] {
+  def column: Column
+  def isColumn: Boolean
+  def isValue: Boolean
```

Comment:
about the isColumn and isValue thing - I would strongly consider implementing this as Either, not via having 2 values that are kind of exclusive

Reply:
`Either` of what. If, then I would make it an enum.
But for now, I actually remove it completely. Can be added if needed later on.